test/e2e/install/install_test.go

```diff
@@ -39,12 +39,12 @@ var testEnv *envtest.Environment
 /// The assumption is that the test-registry is accessible via the IP addr of the master,
 /// or any node in the cluster if the master noe does not exist
 /// TODO Refine how we workout the address of the test-registry
-func getTestClusterDetails() (string, int, []string, error) {
+func getTestClusterDetails() (string, string, int, []string, error) {
```

Comment:
This is getting a bit ridiculous. I would write please consider using struct, but it would probably require doing that all around the code :frowning_face: I think it makes sense to do it at once.

Reply:
there is also ongoing discussion about using some kind of configuration file instead of env variables. Then we would parse yaml/json into a structure and use it internally in the code. But I'm golang noob hence doing only minimal changes and leaving bigger refactorings to others :-)

Reply:
Yeah, I don't think this is about config file (you should parse e.g. json to pre-defined structures in golang otherwise you'll get into stuff like `config.(map[string]interface{})["key1"].(map[string]interface{})["key2"].(string)` which you do not want to, believe me :-) But I agree with the reasoning, actually Chris was kind and did some refactoring in different mr merging this to a struct. But I think I now understand your motivation behind the config file (which I did not really).